src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/ProviderHandler.java

```diff
@@ -45,7 +45,7 @@
      *
      * @throws IllegalArgumentException when no provider is not registered with the given token.
      */
-    public static IRequestResolverProvider getProvider(final IStandardRequestManager manager, final IToken token) throws IllegalArgumentException
+    public static IRequestResolverProvider getProvider(final IStandardRequestManager manager, final IToken<?> token) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)